redfish-core/lib/pcie_slots.hpp

```diff
@@ -90,6 +90,55 @@ inline std::string analysisSlotType(const std::string& slotType)
     return "";
 }
 
+inline void
+    getLinkedPcieDevice(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                        const std::string& slotPath, size_t index)
+{
+    // Collect device associated with this slot and
+    // populate it here
+    crow::connections::systemBus->async_method_call(
+        [asyncResp, index](
+            const boost::system::error_code ec,
+            const std::vector<std::pair<
+                std::string,
+                std::vector<std::pair<std::string, std::vector<std::string>>>>>&
+                subtree) {
+            if (ec)
+            {
+                BMCWEB_LOG_ERROR << "D-Bus response error on GetSubTree " << ec;
+                messages::internalError(asyncResp->res);
+                return;
+            }
+            if (subtree.size() == 0)
+            {
+                BMCWEB_LOG_ERROR
+                    << "Can't find PCIeDevice D-Bus object for given slot";
+                return;
+            }
+
+            // There should be only one path under the slot.
```

Comment:
technically this is true on Rainier, but in general it is not and there is really no value in enforcing it is there?  Update the comment to say "Assuming one device per slot for the moment" so someone looking at it doesn't get confused and think that multiple pcie devices in pcie slots is illegal from a pcie or redfish point of view.

Reply:
updated